### ISSUE TITLE
Linux: pick method of proper type after picking in Explorer

### ISSUE THREAD

**Issue by akavel** [luna/luna-studio#719]
There's weird difference in behavior of Explorer for me on Linux vs on Windows. On Windows, if I type e.g. "get" in Explorer, then hit Enter or double-click on `Http.get`, the app builds properly. On Linux, in the same case the app reports `Runtime error: Can't find function get` on the node. I'm currently on Linux and the source that gets built looks like below:

```
import Std.HTTP
«0»def main:
    «2»get1 = get
    «3»get2 = get
    «1»hello = "Hello, World!"
    None### META {"metas":[{"marker":2,"meta":{"_displayResult":false,"_selectedVisualizer":null,"_position":{"fromPosition":{"_vector2_y":-64,"_vector2_x":-592}}}},{"marker":3,"meta":{"_displayResult":false,"_selectedVisualizer":null,"_position":{"fromPosition":{"_vector2_y":-64,"_vector2_x":-320}}}},{"marker":1,"meta":{"_displayResult":false,"_selectedVisualizer":["base: json","base/json/json.html"],"_position":{"fromPosition":{"_vector2_y":0,"_vector2_x":0}}}}]}
```

Both get1 and get2 report the same exception. I have to type full `Http.get` in Explorer from keyboard to make it work, see get3 below:

```
import Std.HTTP
«0»def main:
    «2»get1 = get
    «4»get3 = Http.get
    «3»get2 = get
    «1»hello = "Hello, World!"
    None### META {"metas":[{"marker":2,"meta":{"_displayResult":false,"_selectedVisualizer":null,"_position":{"fromPosition":{"_vector2_y":-64,"_vector2_x":-592}}}},{"marker":4,"meta":{"_displayResult":false,"_selectedVisualizer":null,"_position":{"fromPosition":{"_vector2_y":80,"_vector2_x":-576}}}},{"marker":3,"meta":{"_displayResult":false,"_selectedVisualizer":null,"_position":{"fromPosition":{"_vector2_y":-64,"_vector2_x":-320}}}},{"marker":1,"meta":{"_displayResult":false,"_selectedVisualizer":["base: json","base/json/json.html"],"_position":{"fromPosition":{"_vector2_y":0,"_vector2_x":0}}}}]}
```

**Comment by mikusp** [comment]
I believe that it is not possible for either `get1` or `get2` to work and you observed some stale result or other bug.

The reason is that `get` is a method, not a function. It needs to be called on some instance of Http class, like `Http.get`. Here `Http` is a constructor of a class that creates a new object containing a `get` method. Otherwise, interpreter tries to find a function `get` that isn't defined anywhere. 
I see that methods and their differences from functions is not properly documented in our user guide. We will improve explanation of this topic.

Thank you for reporting this issue. If you have further questions, I will leave this issue open.

**Comment by akavel** [comment]
The problem is what the Explorer puts into the code.

On Windows, it seems to properly put a `Http.get` after picking a line from Explorer.

On Linux, although i click/Enter over `Http.get`, a raw `get` is picked by Explorer instead.

**Comment by wdanilo** [comment]
It's not exactly connected, but worth linking: https://github.com/luna/luna-studio/issues/607

**Comment by mikusp** [comment]
I checked on Windows today and it doesn't seem to insert `Http.get` like you say. Choosing whether to insert a class constructor when you only typed a method name is a design issue that I think @kustosz is more knowledgeable to resolve

**Comment by akavel** [comment]
Please note, that *I click explicitly on Http.get* in Explorer. I find it hard to understand how this can be interpreted in any other way than that I want explicitly *Http.get* (i.e. *what I clicked*), instead of raw *get* (which *I didn't click*).

**Comment by kustosz** [comment]
@akavel you are absolutely right. There's some issue with the way we display that information if it led you to believe it would add `Http.get`. We'll work on making this cleaner!
